blockmanager.go

```diff
@@ -286,6 +289,15 @@ type blockManagerConfig struct {
 	RpcServer                 func() *rpcServer
 }
 
+// peerSyncState stores additional information that the blockManager tracks
+// about a peer.
+type peerSyncState struct {
+	syncCandidate   bool
+	requestQueue    []*wire.InvVect
```

Comment:
This is unnecessary.  I know it was part of the commit you're porting, but the request logic has since changed which make this no longer needed..